### PR TITLE
Synacor media bidder updates: filter bad sizes & extra video parameters

### PR DESCRIPTION
**Pull request by amuraco**
## Type of change
- [ X ] Feature

## Description of change
- Filter out 1x1 & 1x2 adsizes that we don't support
- Added a bunch of openrtb video params
- Documentation Update: https://github.com/prebid/prebid.github.io/pull/1335

### CODE REVIEW

**Comment by mkendall07** [comment]
This pull request **introduces 1 alert** when merging 5d085eafa4bb943e2ad31eeae1e7e26409606fc4 into 1f9937e1f4503f414f192466d43789409f83e4db - [view on LGTM.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-93f6d9be84f5c7fd238dcb102b5eaa6ba1d35237)

**new alerts:**

* 1 for Implicit operand conversion

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: modules/synacormediaBidAdapter.js [hunk 1]
```diff
@@ -58,33 +64,38 @@ export const spec = {
...
-        };
-        if (videoOrBannerKey === 'video' && bid.params.video) {
-          Object.keys(bid.params.video)
-            .filter(param => includes(VIDEO_PARAMS, param) && !isNaN(parseInt(bid.params.video[param], 10)))
-            .forEach(param => videoOrBannerValue[param] = parseInt(bid.params.video[param], 10));
-        }
-        imp[videoOrBannerKey] = videoOrBannerValue;
-        openRtbBidRequest.imp.push(imp);
-      });
+          let videoOrBannerValue = {
```
Comment (idettman) [inline modules/synacormediaBidAdapter.js]: Since `videoOrBannerValue` does not get reassigned, I’d use `const` rather than `let`

Comment (amuraco) [reply modules/synacormediaBidAdapter.js]: Turns out most of the places with `let` could actually be `const` I've updated the adapter.

**Comment by mkendall07** [comment]
This pull request **introduces 1 alert** when merging 1642e4c78b4150067849ca7be72713aae1c87fef into cacb6e7deb7ac75b33048b3e2cc176be9f980875 - [view on LGTM.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-69937c12856a296549a1da3c89214519adf763dc)

**new alerts:**

* 1 for Implicit operand conversion

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: modules/synacormediaBidAdapter.js [hunk 2]
```diff
@@ -99,8 +110,14 @@ export const spec = {
       };
     }
   },
+
+  setValidVideoParams: function (sourceObj, destObj) {
+    Object.keys(sourceObj)
+      .filter(param => includes(VIDEO_PARAMS, param) && sourceObj[param] !== null && (!isNaN(parseInt(sourceObj[param], 10)) || !sourceObj[param].length < 1))
```
Comment (idettman) [inline modules/synacormediaBidAdapter.js]: The following line:
`!sourceObj[param].length < 1`, is type coercing  to either: `true < 1` or `false < 1` 

Should probably be:
`!(sourceObj[param].length < 1)`

Comment (amuraco) [reply modules/synacormediaBidAdapter.js]: Updated PR with that change, good catch!

**Comment by amuraco** [comment]
I have made the requested changes, is it possible this can be merged for the next release?

**Comment by amuraco** [comment]
Bumping this PR, since all requested changes have been made, and no reply or resolution after 2 weeks

**Pull request merged**